src/xr/xr-hand.js

```diff
@@ -0,0 +1,411 @@
+ * @property {pc.XrHand} hand Hand that joint relates to
+ * @property {pc.XrFinger|null} finger Finger that joint relates to
+ * @property {boolean} wrist True if joint is a wrist
+ * @property {boolean} tip True if joint is a tip of a finger
+ */
+function XrJoint(index, id, hand, finger) {
+    this._index = index;
+    this._id = id;
+
+    this._hand = hand;
+    this._hand._joints.push(this);
+    this._hand._jointsById[id] = this;
+
+    this._finger = finger || null;
+    if (this._finger) this._finger._joints.push(this);
+
+    this._wrist = id == XRHand.WRIST;
+    if (this._wrist) this._hand._wrist = this;
+
+    this._tip = this._finger && !! tipJointIdsIndex[id];
+    if (this._tip) {
+        this._hand._tips.push(this);
+        if (this._finger) this._finger._tip = this;
+    }
+
+    this._radius = null;
+
+    this._localTransform = new Mat4();
+    this._localTransform.setIdentity();
+
+    this._worldTransform = new Mat4();
+    this._worldTransform.setIdentity();
+
+    this._localPosition = new Vec3();
+    this._localRotation = new Quat();
+
+    this._position = new Vec3();
+    this._rotation = new Quat();
+
+    this._dirtyLocal = true;
+}
+
+XrJoint.prototype.update = function (pose) {
+    this._dirtyLocal = true;
+    this._radius = pose.radius;
+    this._localPosition.copy(pose.transform.position);
+    this._localRotation.copy(pose.transform.orientation);
+};
+
+XrJoint.prototype._updateTransforms = function () {
+    var dirty;
+
+    if (this._dirtyLocal) {
+        dirty = true;
+        this._dirtyLocal = false;
+        this._localTransform.setTRS(this._localPosition, this._localRotation, Vec3.ONE);
+    }
+
+    var manager = this._hand._manager;
+    var parent = manager.camera.parent;
+
+    if (parent) {
+        dirty = dirty || parent._dirtyLocal || parent._dirtyWorld;
+        if (dirty) this._worldTransform.mul2(parent.getWorldTransform(), this._localTransform);
+    } else {
+        this._worldTransform.copy(this._localTransform);
+    }
+};
+
+/**
+ * @function
+ * @name pc.XrJoint#getPosition
+ * @description Get the world space position of a joint
+ * @returns {pc.Vec3} The world space position of a joint
+ */
+XrJoint.prototype.getPosition = function () {
+    this._updateTransforms();
+    this._worldTransform.getTranslation(this._position);
+    return this._position;
+};
+
+/**
+ * @function
+ * @name pc.XrJoint#getRotation
+ * @description Get the world space rotation of a joint
+ * @returns {pc.Quat} The world space rotation of a joint
+ */
+XrJoint.prototype.getRotation = function () {
+    this._updateTransforms();
+    this._rotation.setFromMat4(this._worldTransform);
+    return this._rotation;
+};
+
+/**
+ * @function
+ * @name pc.XrJoint#getRadius
+ * @description Get the radius of a joint, which is a distance from joint to the edge of a skin
+ * @returns {number} Radius of a joint
+ */
+XrJoint.prototype.getRadius = function () {
```

Comment:
I would suggest making this a read-only property.

Reply:
This is important value for developer, to for example making physical capsules for fingers. Joint radius is different for different people. And is calculated by computer vision.

Reply:
Sure, I get it's important. I'm just suggesting it should be `pc.XrJoint#radius` instead of `pc.XrJoint#getRadius`. What do you think?

Reply:
Accessor, sure, can be.